iroh/src/node.rs

```diff
@@ -1480,6 +1481,79 @@ impl<D: BaoStore> RpcHandler<D> {
         rx.into_stream()
     }
 
+    fn blob_read_at(
+        self,
+        req: BlobReadAtRequest,
+    ) -> impl Stream<Item = RpcResult<BlobReadAtResponse>> + Send + 'static {
+        let (tx, rx) = flume::bounded(RPC_BLOB_GET_CHANNEL_CAP);
+        let entry = self.inner.db.get(&req.hash);
+        self.inner.rt.spawn_pinned(move || async move {
+            if let Err(err) = read_loop(
+                req.offset,
+                req.len,
+                entry,
+                tx.clone(),
+                RPC_BLOB_GET_CHUNK_SIZE,
+            )
+            .await
+            {
+                tx.send_async(RpcResult::Err(err.into())).await.ok();
+            }
+        });
+
+        async fn read_loop<M: Map>(
+            offset: u64,
+            len: Option<usize>,
+            entry: Option<impl MapEntry<M>>,
+            tx: flume::Sender<RpcResult<BlobReadAtResponse>>,
+            max_chunk_size: usize,
```

Comment:
Do we need the max_chunk_size? Makes the code more complex.

At least in the AsyncSliceReader the idea is that you would only ask for sizes that you are fine with handling as a single Bytes. That is why the len parameter is `usize`, not `u64`.

Reply:
it's needed to limit the chunk size sent over the wire , like in the regular read

Reply:
I don't quite get it. The read_at API returns a `Bytes`, so if you call it with some large size, you will at one point get a Bytes containing all the data in memory. So what's the benefit of this chunking for the purpose of implementing AsyncReadAt?

Reply:
the rpc `read_at` api does not return the full `Bytes`, but rather streams it, so no we will not get the full data in memory with the current strategy